db/osqlcomm.c

```diff
@@ -7419,89 +7392,76 @@ static int sorese_rcvreq(char *fromhost, void *dtap, int dtalen, int type,
         debug = 1;
     }
 
-    /*
-       Blockproc does not create a copy of the request,
+    /* Blockproc does not create a copy of the request,
        but creates a different thread to work on it
-       Let THAT thread free it...
-    free(p_buf);
-    */
+       Let THAT thread free p_buf (malcd)...  */
```

Comment:
to answer the above comment, malcd needs to survive...see this line.

Reply:
thanks - we can try passing it in so don't have to get from heap

Reply:
also simplifies cleanup on error

Reply:
good idea, will try in separate PR.